substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/headers/darwin/DarwinStat.java

```diff
@@ -41,16 +41,16 @@
 public class DarwinStat {
 
     @CStruct(addStructKeyword = true)
-    public interface stat64 extends PointerBase {
+    public interface stat extends PointerBase {
         @CField
         long st_size();
     }
 
-    @CFunction("fstat64")
-    public static native int fstat64(int fd, stat64 buf);
+    @CFunction("fstat")
```

Comment:
The situation is a bit more complicated than that: `_DARWIN_FEATURE_ONLY_64_BIT_INODE`, which is the default by now, changes the entire `struct stat` layout, which we notice because we get the struct layout from the header files (via generated C code, see `QueryCodeWriter`). But the header files also change C calls from `fstat` to `fstat$INODE64`, which we don't notice because what we specify in `@CFunction` is the actual symbol name. I have the following patch to address this (and explicitly define `_DARWIN_USE_64_BIT_INODE` to be safe). Can you please check that this works for the app store?
```patch
diff --git a/substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/headers/PosixDirectives.java b/substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/headers/PosixDirectives.java
index 14eb719e94f..8c877c6328e 100644
--- a/substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/headers/PosixDirectives.java
+++ b/substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/headers/PosixDirectives.java
@@ -95,6 +95,6 @@ public class PosixDirectives implements CContext.Directives {
 
     @Override
     public List<String> getMacroDefinitions() {
-        return Arrays.asList("_GNU_SOURCE", "_LARGEFILE64_SOURCE");
+        return Arrays.asList("_GNU_SOURCE", "_LARGEFILE64_SOURCE", "_DARWIN_USE_64_BIT_INODE");
     }
 }
diff --git a/substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/headers/darwin/DarwinStat.java b/substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/headers/darwin/DarwinStat.java
index de1296e69e7..0890dfef134 100644
--- a/substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/headers/darwin/DarwinStat.java
+++ b/substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/headers/darwin/DarwinStat.java
@@ -40,17 +40,23 @@ import com.oracle.svm.core.posix.headers.PosixDirectives;
 @CContext(PosixDirectives.class)
 public class DarwinStat {
 
+    /*
+     * NOTE that we set _DARWIN_USE_64_BIT_INODE in the C directives to force a layout for struct
+     * stat with a 64-bit st_ino, and we have to call functions with a $INODE64 suffix to match,
+     * such as fstat$INODE64.
+     */
+
     @CStruct(addStructKeyword = true)
     public interface stat extends PointerBase {
         @CField
         long st_size();
     }
 
-    @CFunction("fstat")
+    @CFunction("fstat$INODE64")
     public static native int fstat(int fd, stat buf);
 
     public static class NoTransitions {
-        @CFunction(transition = CFunction.Transition.NO_TRANSITION)
+        @CFunction(value = "fstat$INODE64", transition = CFunction.Transition.NO_TRANSITION)
         public static native int fstat(int fd, stat buf);
     }
 }
```